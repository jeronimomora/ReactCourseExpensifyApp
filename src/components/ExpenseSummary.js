import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux'
import getExpenseTotal from '../selectors/expense-total'
import getVisibleExpenses from '../selectors/expenses'

export const ExpenseSummary = (props) => {
	return (<div>
	{
			props.expensesCount > 0 ?
		(
			<h1>Viewing {props.expensesCount} {' '}
			{props.expensesCount > 1 ? 'expenses' : 'expense'} totalling {' '}
			{numeral(props.expensesTotal/100).format('$0,0.00')}</h1>
		)
		 : (<h1>Viewing 0 expenses totalling $0.00</h1>)
		
	}</div>)
}

const mapStateToProps = (state) => {
	const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
	return {
		expensesCount: visibleExpenses.length,
		expensesTotal: getExpenseTotal(visibleExpenses)
	}
}

export default connect(mapStateToProps)(ExpenseSummary)