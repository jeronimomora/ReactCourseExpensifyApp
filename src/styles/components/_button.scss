.button {
	background: $light-blue;
	border: none;
	border-radius: 2px;
	color: white;
	display: inline-block;
	line-height: 1;
	font-weight:  300;
	font-size: $font-size-large;
	padding: $s-size;
	text-decoration: none;
}

.button--link {
	background: none;
}

.button--secondary {
	background: #888888;
	margin-bottom: 5;
}