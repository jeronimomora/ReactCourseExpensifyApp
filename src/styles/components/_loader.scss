.loader {
	height: 100vh;
	width: 100vw;
	display: flex;
	align-items: center;
	justify-content: center;
}

.loader__image {
	height: 6 rem;
	width: 6 rem;
}