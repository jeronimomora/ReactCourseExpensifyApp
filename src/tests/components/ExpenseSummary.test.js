import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'
import expenses from '../fixtures/expenses'

test('snapshot for one expense', () => {
	const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={10000}/>)
	expect(wrapper).toMatchSnapshot()
})

test('snapshot for two expenses', () => {
	const wrapper = shallow(<ExpenseSummary expensesCount={2} expensesTotal={20000}/>)
	expect(wrapper).toMatchSnapshot()
})