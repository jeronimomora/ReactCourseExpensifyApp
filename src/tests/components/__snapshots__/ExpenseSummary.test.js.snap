// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
     
    expense
     totalling 
     
    $100.00
  </h1>
</div>
`;

exports[`snapshot for two expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
     
    expenses
     totalling 
     
    $200.00
  </h1>
</div>
`;
